core/templates/pages/exploration-editor-page/settings-tab/settings-tab.component.ts

```diff
@@ -218,7 +218,9 @@ export class SettingsTabComponent
   }
 
   updateCategoryListWithUserData(): void {
-    if (this.newCategory) {
+    if (this.newCategory &&
+      (this.explorationCategoryService.displayed === this.newCategory.id)
```

Comment:
This looks like a hack. Could you explain please do we need this clause?

Reply:
This just makes sure that the newly selected category is equal to the new category, otherwise it won't create a new category. Beforehand if you were to search up a category, it would create a new one regardless because filterChoices was setting the new category and this clause would always be true.